src/module.c

```diff
@@ -7471,6 +7485,18 @@ void processModuleLoadingProgressEvent(int is_aof) {
     }
 }
 
+/* When a module key is deleted (in dbAsyncDelete/dbSyncDelete), it will be called 
+ * to tell the module which key is about to be released */
+void moduleNotifyKeyUnlink(robj *key, robj *val) {
+    if (val->type == OBJ_MODULE) {
+        moduleValue *mv = val->ptr;
+        moduleType *mt = mv->type;
+        if (mt->unlink != NULL) {
+            mt->unlink(key);
```

Comment:
maybe we want to let the module have access to the `mt` (`val`) too?
some modules may want to look into some variable they have there, and be forced to do RM_OpenKey for that.

another alternative is maybe pass a `RedisModuleKey` and then they can get the key name via RM_GetKeyNameFromModuleKey, and value via RM_ModuleTypeGetValue

what do you guys thing?

Reply:
Well, If necessary, I'd prefer to just pass `val` over because it's already there.

Reply:
@guybe7 @MeirShpilraien @yossigo what do you think?
i suppose a void* value is consistent with other APIs:
```c
typedef void (*RedisModuleTypeDigestFunc)(RedisModuleDigest *digest, void *value);
typedef void (*RedisModuleTypeUnlinkFunc)(RedisModuleString *key, const void *value);
```

i'm ok with this, but i want to give it a bit more thought so that we don't regret it in the future.

The alternative of providing a `RedisModuleKey*` will let the user get the key name, value and also other things (like LRU) from it.

Reply:
@oranagra I lean towards just passing the value, I don't see any real need to manipulate the key (after all it's getting unlinked) and it's more consistent with the other calls.